entity/dag/clock.go

```diff
@@ -18,21 +19,13 @@ func ClockLoader(defs ...Definition) repository.ClockLoader {
 	return repository.ClockLoader{
 		Clocks: clocks,
 		Witnesser: func(repo repository.ClockedRepo) error {
-			// we need to actually load the identities because of the commit signature check when reading,
-			// which require the full identities with crypto keys
-			resolver := identity.NewCachedResolver(identity.NewSimpleResolver(repo))
-
+			var errG errgroup.Group
 			for _, def := range defs {
-				// we actually just need to read all entities,
-				// as that will create and update the clocks
-				// TODO: concurrent loading to be faster?
-				for b := range ReadAll(def, repo, resolver) {
-					if b.Err != nil {
-						return b.Err
-					}
-				}
+				errG.Go(func() error {
```

Comment:
Bonus here: parallelize and only read the necessary data to build the clocks. On my machine, speed up this operation from 210ms to 120ms.

Reply:
🚀